packages/xo-server/src/index.mjs

```diff
@@ -149,6 +149,10 @@ async function createExpressApp(config) {
   // access the username and password from the sign in form.
   app.use(bodyParser.urlencoded({ extended: false }))
 
+  // Registers the body-parser json middleware, needed to retrieve
+  // the body of the POST/PUT request in `req.body`
+  app.use(bodyParser.json())
```

Comment:
I'm not fond of this either because it applies to all the requests.

Can you add it as middleware to the specific route instead? (simply pass the middleware before the route handler).